code/datums/quirks/negative.dm

```diff
@@ -1075,3 +1074,48 @@
 	if(!IS_ORGANIC_LIMB(old_limb))
 		cybernetics_level--
 		update_mood()
+
+/datum/quirk/decrepit
+	name = "Decrepit"
+	desc = "You require a cane in your hand for mobility."
+	icon = "person-cane"
+	value = -4
+	hardcore_value = 4
+	gain_text = "<span class='danger'>You feel old!</span>"
+	lose_text = "<span class='notice'>You feel young again.</span>"
+	medical_record_text = "Patient suffers from a lack of mobility requiring a cane for support."
+	mail_goodies = list(/obj/item/cane)
+
+/datum/quirk/decrepit/add_unique()
+	var/turf/holder_turf = get_turf(quirk_holder)
+	var/obj/item/cane/spawn_cane = new /obj/item/cane(holder_turf)
+	quirk_holder.put_in_hands(spawn_cane)
+	RegisterSignal(quirk_holder, COMSIG_MOVABLE_MOVED, PROC_REF(handle_decrepit_movement))
+
+/datum/quirk/decrepit/remove()
+	UnregisterSignal(quirk_holder, list(COMSIG_MOVABLE_MOVED))
+
+/datum/quirk/decrepit/proc/handle_decrepit_movement(datum/source, old_loc)
+	SIGNAL_HANDLER
+
+	if(quirk_holder.stat != CONSCIOUS || quirk_holder.IsSleeping() || quirk_holder.IsUnconscious() || quirk_holder.body_position == LYING_DOWN || quirk_holder.buckled )
+		return
+
+	if(quirk_holder.m_intent == MOVE_INTENT_WALK)
+		return
+
+	for(var/obj/item/held_cane in quirk_holder.held_items)
+		if(HAS_TRAIT(held_cane, TRAIT_CANE_TOOL))
+			return
+
+	switch(rand(0, 100))
+		if(0 to 70)
+			return
+		if(71 to 90)
+			to_chat(quirk_holder, span_warning("You feel off balance without your cane. It might be safer to walk..."))
```

Comment:
This should probably be a status alert so that it's more obvious, only giving chat feedback is bad if you're not already looking at chat.

Not suggesting balloon alert here since it's important enough to not be forgotten, but also because it means that it can also go away once you equip a cane again.